### ISSUE TITLE
[Issue]: Albums not displaying correctly, including artist names/genres missing, discs not separated by Jellyfin, and track number included in title

### ISSUE THREAD

**Issue by suchasaltylemon** [jellyfin/jellyfin#9359]
### Please describe your bug

This issue I have found to occur on 10.8.9, but, after downgrading to 10.8.7, the issue does not occur. I have not tested 10.8.8.

When importing an album, manually or automatically (Lidarr), the album is imported and can be played, but the display format is not as expected. This involves:

- Artist name missing from under album cover under the 'Albums' section of the music explorer
- Albums not being separated into their separate discs when appropriate
- Track titles including the track number
- Track number not displaying in its own column
- Genres not being displayed

This occurs for all new albums on this version. Each audio file follows the convention as outlined [in the docs](https://jellyfin.org/docs/general/server/media/music/). They have all been tagged using [MusicBrainz Picard](https://picard.musicbrainz.org/) and the tags have been confirmed by [Mp3tag](https://www.mp3tag.de/en/). 

Upon removing the library from Jellyfin and readding it, all previously correct albums (from older versions of Jellyfin) lose the correct display.

The issue is fixed for all old albums when reverting to 10.8.7, refreshing all metadata, and then upgrading to 10.8.9 again, though new albums also experience the same issue.

This issue was initially discussed on [reddit](https://www.reddit.com/r/jellyfin/comments/114wn9k/jellyfin_not_setting_correct_track_number_or/), before discovering this was a bug.

### Jellyfin Version

Other

### if other:

10.8.9

### Environment

```markdown
- OS: Windows 10 x64
- Virtualization: None
- Client: Browser
- Browser: Firefox 110.0
- Hardware Acceleration: NVENC
- Plugins: AudioDB, FanArt, MusicBrainz, MergeVersions, OMDb, OpenSubtitles, Playback Reporting, Reports, SkinManager, StudioImages, TMDb, TMDb Box Sets, TMDb Trailers, TheTVDB
- Reverse Proxy: Caddy
- Base URL: None
- Storage: Local
```

### Please attach any screenshots here

Album cover missing artist name underneath
![image](https://user-images.githubusercontent.com/46686300/219974103-375164be-553a-413d-8596-4bdb729a7cbf.png) 

Album cover should display like this
![image](https://user-images.githubusercontent.com/46686300/219974199-b7efae2b-63d4-49cc-9170-25437630b54c.png)

Album not separating into discs or separating track number from track name
![image](https://user-images.githubusercontent.com/46686300/219974115-7ee1a7d8-4c05-44f9-9cb5-5eb57ceb0975.png)

Expected album display
![image](https://user-images.githubusercontent.com/46686300/219974168-749fa768-6711-4d82-9c0e-23b6f74f16f4.png)

'evermore' folder structure
![image](https://user-images.githubusercontent.com/46686300/219974126-2d63f9ca-68bf-4aba-833d-1df0039d9f7a.png)

'good kid, m.A.A.d city' folder structure
![image](https://user-images.githubusercontent.com/46686300/219974239-97bebde7-1e8b-4770-9c67-218e0c5c4af1.png)

'evermore' metadata (as shown by Mp3tag)
![image](https://user-images.githubusercontent.com/46686300/219974273-ab4c39b9-983c-4fba-8068-d32ce9e9d660.png)

'good kid, m.A.A.d city' metadata (as shown by Mp3tag)
![image](https://user-images.githubusercontent.com/46686300/219974300-c66c306d-8c56-42f3-b443-58906f389b1f.png)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by MdXMaxX** [comment]
I believe I'm experiencing this same thing, or at least something very similar.

I had no issues with multi-disc albums previously.  However, I went to add a couple of multi-disc albums to one of my libraries today, and Jellyfin decides not to load the album art or group the tracks by disc for either album.  Refreshing the metadata and the library were no help.

I decided to remove the library and re-add it, and unfortunately the result of that is *every* multi-disc album in this library now lacks album art.  *Some* of these albums seem to have their tracks grouped by disc properly, but many do not.  Really unfortunate as this library now just looks like a total mess.

Unfortunately, downgrading to 10.8.7 did not resolve this for me, I see the same behavior after removing and re-adding the library again.

I suspect this has been an issue for several versions.  I found #8396 which seems to describe the exact same thing, and that was reported on 10.8.4.

Why I haven't experienced this until now, I'm not sure.  Perhaps I just haven't added a multi-disc album to my library in a while.

I'll try downgrading to 10.7.7, which was the latest release when I started using Jellyfin about a year ago, and see if that manages to resolve the issue.

**Comment by MdXMaxX** [comment]
I figured out another workaround that isn't ideal but I can live with for now until this bug is resolved.

We need to
a) Ensure the "Discnumber" tag is populated in the metadata of the music files
b) Organize the music files by filename instead of splitting them into folders

For example, converting this:
```
/Album
  /Disc 1
    /01 - Song 1.mp3
    /02 - Song 2.mp3
  /Disc 2
    /01 - Song 1.mp3
    /02 - Song 2.mp3
```
to this:
```
/Album
  /1-01 - Song 1.mp3
  /1-02 - Song 2.mp3
  /2-01 - Song 1.mp3
  /2-02 - Song 2.mp3
```

**Comment by MdXMaxX** [comment]
This issue is still present.  I'm eagerly waiting for #10029 to make its way into a Jellyfin release; it *seems* like it will fix my issues.

**Comment by mihawk90** [comment]
> * Albums not being separated into their separate discs when appropriate

This seems to be fixed in the meantime, at least it works on 10.8.11

Unfortunately, it does not show [disc titles](https://musicbrainz.org/doc/Medium#:~:text=an%20optional%20title%20(e.g.%20disc%202%3A%20The%20Early%20Years)), but that is probably more a feature request then a bug.

IMO Jellyfin relying on folder structure at all is neither intuitive nor constructive at all. Metadata should always trump folder structure. In my eyes the only acceptable situation for relying on directory structure is if there is no or insufficient Metadata available. But that's probably an entirely different discussion and would require a major rework of the import process overall.
